### PR TITLE
Added staging values for queues and storage account

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1217

### Change description ###
• This PR overrides queue connection string and storage secrets with staging infra ones.
• Other secrets remain same but needs to be present while overriding.
• This PR is dependent on hmcts/bulk-scan-shared-infrastructure#169 and hmcts/bulk-scan-shared-infrastructure#171 to create secrets in key vault.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/bulk-scan-processor/values.aat.template.yaml [hunk 1]
```diff
@@ -1,4 +1,19 @@
 java:
+  keyVaults:
+    "bulk-scan":
+      secrets:
+        - processor-POSTGRES-PASS
+        - s2s-secret
+        - flyway-password
+        - envelopes-staging-queue-send-connection-string
+        - notifications-queue-send-connection-string
+        - processed-envelopes-queue-listen-connection-string
```
Comment (mdayican) [inline charts/bulk-scan-processor/values.aat.template.yaml:10]: these should be staging ?

File: charts/bulk-scan-processor/values.yaml [hunk 2]
```diff
@@ -24,10 +24,7 @@ java:
...
     STORAGE_BLOB_PUBLIC_KEY: "nonprod_public_key.der"
     STORAGE_URL: https://bulkscan.{{ .Values.global.environment }}.platform.hmcts.net
-    BULK_SCANNING_DB_USER_NAME: bulk_scanner@bulk-scan-processor-{{ .Values.global.environment }}
-    BULK_SCANNING_DB_NAME: bulk_scan
-    BULK_SCANNING_DB_HOST: bulk-scan-processor-{{ .Values.global.environment }}.postgres.database.azure.com
-    BULK_SCANNING_DB_PORT: "5432"
+    FLYWAY_SKIP_MIGRATIONS: true
     FLYWAY_URL: jdbc:postgresql://bulk-scan-processor-{{ .Values.global.environment }}.postgres.database.azure.com:5432/bulk_scan?sslmode=require
     FLYWAY_USER: bulk_scanner@bulk-scan-processor-{{ .Values.global.environment }}
     FLYWAY_NOOP_STRATEGY: "true"
```
Comment (doncem) [inline charts/bulk-scan-processor/values.yaml:30]: oh, actually this one uses some older config called `FLYWAY_NOOP_STRATEGY`. need a change ;)

Comment (doncem) [reply charts/bulk-scan-processor/values.yaml:30]: or if you want to - can use this and remove the old one. but then need a change in src code

File: build.gradle [hunk 3]
```diff
@@ -284,14 +284,6 @@ bootJar {
 
 import org.flywaydb.gradle.task.FlywayMigrateTask
 
-flyway {
```
Comment (nitinprabhu) [inline build.gradle:287]: is this change part of this PR ?

File: charts/bulk-scan-processor/Chart.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 name: bulk-scan-processor
 apiVersion: v1
 home: https://github.com/hmcts/bulk-scan-processor
-version: 0.3.0
+version: 0.3.1
```
Comment (nitinprabhu) [inline charts/bulk-scan-processor/Chart.yaml:4]: why we need to bump version ?

File: charts/bulk-scan-processor/values.preview.template.yaml [hunk 5]
```diff
@@ -29,9 +29,8 @@ java:
     BULK_SCANNING_DB_PASSWORD: "{{ .Values.postgresql.postgresqlPassword}}"
     BULK_SCANNING_DB_NAME: "{{ .Values.postgresql.postgresqlDatabase}}"
     BULK_SCANNING_DB_HOST: "{{ .Release.Name }}-postgresql"
-    FLYWAY_URL: "jdbc:postgresql://{{ .Release.Name }}-postgresql:5432/{{ .Values.postgresql.postgresqlDatabase}}"
-    FLYWAY_USER: "{{ .Values.postgresql.postgresqlUsername}}"
-    FLYWAY_PASSWORD: "{{ .Values.postgresql.postgresqlPassword}}"
+    BULK_SCANNING_DB_PORT: '5432'
+    BULK_SCANNING_DB_CONN_OPTIONS: ""
```
Comment (nitinprabhu) [inline charts/bulk-scan-processor/values.preview.template.yaml:33]: not part of this PR ?

File: charts/bulk-scan-processor/values.yaml [hunk 6]
```diff
@@ -44,6 +44,10 @@ java:
     "bulk-scan":
       secrets:
         - processor-POSTGRES-PASS
+        - processor-POSTGRES-USER
+        - processor-POSTGRES-DATABASE
+        - processor-POSTGRES-HOST
+        - processor-POSTGRES-PORT
```
Comment (nitinprabhu) [inline charts/bulk-scan-processor/values.yaml:50]: why do we need to add this for staging ?

Comment (apereverzin) [reply charts/bulk-scan-processor/values.yaml]: @nitinprabhu Oh, yes

File: src/main/resources/bootstrap.yaml [hunk 7]
```diff
@@ -5,13 +5,27 @@ spring:
       paths: /mnt/secrets/bulk-scan
       aliases:
         processor-POSTGRES-PASS: BULK_SCANNING_DB_PASSWORD
+        processor-POSTGRES-USER: BULK_SCANNING_DB_USER_NAME
+        processor-POSTGRES-DATABASE: BULK_SCANNING_DB_NAME
+        processor-POSTGRES-HOST: BULK_SCANNING_DB_HOST
+        processor-POSTGRES-PORT: BULK_SCANNING_DB_PORT
```
Comment (nitinprabhu) [inline src/main/resources/bootstrap.yaml:11]: Not sure why we need those when we already have it defined https://github.com/hmcts/bulk-scan-processor/blob/master/charts/bulk-scan-processor/values.yaml#L27-L30

File: src/main/resources/bootstrap.yaml [hunk 8]
```diff
@@ -5,13 +5,22 @@ spring:
       paths: /mnt/secrets/bulk-scan
       aliases:
         processor-POSTGRES-PASS: BULK_SCANNING_DB_PASSWORD
+        processor-staging-db-password: BULK_SCANNING_DB_PASSWORD
+        processor-staging-db-user: BULK_SCANNING_DB_USER_NAME
+        processor-staging-db-name: BULK_SCANNING_DB_NAME
+        processor-staging-db-host: BULK_SCANNING_DB_HOST
+        processor-staging-db-port: BULK_SCANNING_DB_PORT
```
Comment (lgonczar) [inline src/main/resources/bootstrap.yaml:12]: How is it that these mappings are not defined for non-staging DB?

Comment (nitinprabhu) [reply src/main/resources/bootstrap.yaml:12]: good point @lgonczar they are defined in values yaml `https://github.com/hmcts/bulk-scan-processor/blob/master/charts/bulk-scan-processor/values.yaml#L27-L30` so for staging we can do similar in value aat template

File: charts/bulk-scan-processor/values.aat.template.yaml [hunk 9]
```diff
@@ -1,8 +1,31 @@
...
   ingressHost: ${SERVICE_FQDN}
   devApplicationInsightsInstrumentKey: "f666440f-43c0-4abb-9c20-2ce175a374f0"
   environment:
     STORAGE_BLOB_SELECTED_CONTAINER: "ALL"
     LEASE_ACQUIRE_DELAY_IN_SECONDS: "15"
+    FLYWAY_NOOP_STRATEGY: "false"
+    BULK_SCANNING_DB_USER_NAME: bulk_scanner@bulk-scan-processor-staging-{{ .Values.global.environment }}
+    BULK_SCANNING_DB_NAME: bulk_scan
+    BULK_SCANNING_DB_HOST: bulk-scan-processor-staging-{{ .Values.global.environment }}.postgres.database.azure.com
+    BULK_SCANNING_DB_PORT: "5432"
```
Comment (nitinprabhu) [inline charts/bulk-scan-processor/values.aat.template.yaml:31]: port is same so need not be overrided

File: charts/bulk-scan-processor/values.aat.template.yaml [hunk 10]
```diff
@@ -1,8 +1,31 @@
...
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
   devApplicationInsightsInstrumentKey: "f666440f-43c0-4abb-9c20-2ce175a374f0"
   environment:
     STORAGE_BLOB_SELECTED_CONTAINER: "ALL"
     LEASE_ACQUIRE_DELAY_IN_SECONDS: "15"
+    FLYWAY_NOOP_STRATEGY: "false"
+    BULK_SCANNING_DB_USER_NAME: bulk_scanner@bulk-scan-processor-staging-{{ .Values.global.environment }}
+    BULK_SCANNING_DB_NAME: bulk_scan
+    BULK_SCANNING_DB_HOST: bulk-scan-processor-staging-{{ .Values.global.environment }}.postgres.database.azure.com
```
Comment (nitinprabhu) [inline charts/bulk-scan-processor/values.aat.template.yaml:30]:
```suggestion
    BULK_SCANNING_DB_USER_NAME: bulk_scanner@bulk-scan-processor-staging-aat
    BULK_SCANNING_DB_HOST: bulk-scan-processor-staging-aat.postgres.database.azure.com
```

Comment (nitinprabhu) [reply charts/bulk-scan-processor/values.aat.template.yaml:30]: Also `BULK_SCANNING_DB_NAME: bulk_scan` is already in values yaml

File: charts/bulk-scan-processor/values.aat.template.yaml [hunk 11]
```diff
@@ -1,8 +1,29 @@
 java:
+  keyVaults:
+    "bulk-scan":
+      secrets:
+        - processor-staging-db-password
+        - processor-staging-db-user
+        - processor-staging-db-name
+        - processor-staging-db-host
+        - processor-staging-db-port
```
Comment (nitinprabhu) [inline charts/bulk-scan-processor/values.aat.template.yaml:9]: should we remove this now as you have removed bootstrap mapping ?
